e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt

```diff
@@ -0,0 +1,218 @@
+    #
+    FLUSH PRIVILEGES;
+"
+
+statement ok
+create source s with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
+  password = '123456',
+  database.name = 'testdb1',
+  server.id = '114514'
+);
+
+sleep 1s
+
+# There's no data in upstream yet, but the offset is updated in the source executor via the heartbeat message.
+system ok
+internal_table.mjs --name s --type '' --count
+----
+count: 1
+
+
+system ok
+mysql -e "
+    INSERT INTO tt1 VALUES (1, '2023-10-23 10:00:00');
+    INSERT INTO tt2 VALUES (2, '2023-10-23 11:00:00');
+    INSERT INTO tt3 VALUES (3, '2023-10-23 12:00:00');
+    INSERT INTO tt4 VALUES (4, '2023-10-23 13:00:00');
+    INSERT INTO tt5 VALUES (5, '2023-10-23 14:00:00');
+" testdb1
+
+statement ok
+create table tt1 (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
+  password = '123456',
+  database.name = 'testdb1',
+  table.name = 'tt1',
+);
+
+statement ok
+create table tt2 (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
+  password = '123456',
+  database.name = 'testdb1',
+  table.name = 'tt2',
+);
+
+statement ok
+create table tt3 (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
+  password = '123456',
+  database.name = 'testdb1',
+  table.name = 'tt3',
+);
+
+statement ok
+create table tt4 (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
+  password = '123456',
+  database.name = 'testdb1',
+  table.name = 'tt4',
+);
+
+statement ok
+create table tt5 (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) with (
+  ${RISEDEV_MYSQL_WITH_OPTIONS_COMMON},
+  username = 'non-shared-cdc',
+  password = '123456',
+  database.name = 'testdb1',
+  table.name = 'tt5',
+);
+
+statement ok
+create table tt1_shared (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) from s table 'testdb1.tt1';
+
+statement ok
+create table tt2_shared (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) from s table 'testdb1.tt2';
+
+statement ok
+create table tt3_shared (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) from s table 'testdb1.tt3';
+
+statement ok
+create table tt4_shared (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) from s table 'testdb1.tt4';
+
+statement ok
+create table tt5_shared (v1 int,
+  v2 timestamptz,
+  PRIMARY KEY (v1)
+) from s table 'testdb1.tt5';
+
+sleep 5s
+
+query IT
+select * from tt1;
+----
+1 2023-10-23 10:00:00+00:00
+
+query IT
+select * from tt2;
+----
+2 2023-10-23 11:00:00+00:00
+
+query IT
+select * from tt3;
+----
+3 2023-10-23 12:00:00+00:00
+
+query IT
+select * from tt4;
+----
+4 2023-10-23 13:00:00+00:00
+
+query IT
+select * from tt5;
+----
+5 2023-10-23 14:00:00+00:00
+
+
+query IT
+select * from tt1_shared;
+----
+1 2023-10-23 10:00:00+00:00
```

Comment:
Hmmm, this test succeeds in CI, but failed for me locally. `tt1` is ok, but `tt1_shared` has a different result. Any ideas?

```
failed to run `e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt`

Caused by:
    query result mismatch:
    [SQL] select * from tt1_shared;
    [Diff] (-expected|+actual)
    -   1 2023-10-23 10:00:00+00:00
    +   1 2023-10-23 02:00:00+00:00
    at e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt:177
    
```

Reply:
cc @xiangjinwu @StrikeW

Reply:
Most likely a bug in test setup or CDC itself, as the result is different in 8 hours when executed in different time zone.

Reply:
In the `MySqlExternalTableReader` which backfill upstream table we have executed `conn.exec_drop("SET time_zone = \"+00:00\"", ()).await?;` before scan the rows. And set `driver.connectionTimeZone=+00:00` in the debezium connector.

Reply:
@StrikeW Sorry that I might not understand the details yet. Do you mean this is a bug or ..?

Reply:
I think it should not happen, is it a new test case you added?

Reply:
Could you try to run this test locally to see whether you can reproduce it?

Reply:
Is it a simple way to run the test locally with your branch?

Reply:
Yes
1. Add your MySQL to risedev user profile
2. risedev d 
3. risedev slt e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt

Reply:
Let's create a separate issue for this
https://github.com/risingwavelabs/risingwave/issues/16496